crates/oxc_semantic/src/jsdoc/mod.rs

```diff
@@ -0,0 +1,46 @@
+mod builder;
+
+use std::{cell::RefCell, rc::Rc};
+
+pub use builder::JsDocBuilder;
+use oxc_ast::{GetSpan, Span};
+use rustc_hash::FxHashMap;
+
+use self::parser::JsDocParser;
+pub use self::parser::JsDocTag;
+use crate::AstNode;
+
+mod parser;
+
+#[derive(Debug)]
+pub struct JsDoc<'a> {
+    /// AstKind Span -> JsDoc Span
+    span_map: FxHashMap<Span, &'a str>,
+
+    /// Cached parsed jsdoc
+    parsed: RefCell<FxHashMap<Span, Rc<[JsDocTag<'a>]>>>,
```

Comment:
`[JsDocTag<'a>]` Should be it's own struct.

Reply:
Something like?

```rust
#[derive(Debug)]
pub struct JsDocTags<'a> {
  tags: Rc<[JsDocTag<'a>]>,
}
```